### PR TITLE
Initial PR

### PR DESCRIPTION
**Pull request by michaelfeil**
Adds initial support. Tested the readme for local deployment.

### CODE REVIEW

**Comment by michaelfeil** [comment]
@ssheng ready for review.

File: bentofile.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+labels:
+  owner: bentoml-team
+  project: gallery
+include:
+  - "*.py"
+python:
+  requirements_txt: requirements.txt
+docker:
+  system_packages:
+    - git
```
Comment (larme) [inline bentofile.yaml]: Why we need git here? Does installing `infinity_emb` require git?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bentoml>=1.2.17
+infinity_emb[all]>=0.0.50
```
Comment (larme) [inline requirements.txt]: As an example project, we should lock `infinity_emb`'s version to 0.0.50

File: service.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+class INFINITY:
+    """
+    This class is inspired by the implementation shown in the infinity project.
+    Source: https://github.com/michaelfeil/infinity
+    """
+
+    def __init__(self):
+        from infinity_emb import AsyncEngineArray, EngineArgs
+
+        self.array = AsyncEngineArray.from_args(
```
Comment (larme) [inline service.py]: I think `self.array` -> `self.engine_array` would be a better name.

File: service.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            [
+                EngineArgs(
+                    model_name_or_path="BAAI/bge-small-en-v1.5",
+                    engine="torch",
+                    embedding_dtype="float32",
+                    batch_size=4,
```
Comment (larme) [inline service.py]: why is `batch_size` set to 4?

File: service.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                EngineArgs(
+                    model_name_or_path="wkcn/TinyCLIP-ViT-8M-16-Text-3M-YFCC15M",
+                    engine="torch",
+                    batch_size=4,
+                ),
+            ]
+        )
+
+    @bentoml.api
+    async def embeddings(self, input: list[str], model: str) -> t.Dict:
```
Comment (larme) [inline service.py:44]: Maybe we should give`input` a more meaningful name (e.g. like `sentences`?)

**Review by larme — CHANGES_REQUESTED** [review]
Overall very good PR, just left some minor suggestions. Also could you give every endpoint's api function some meaningful default inputs so user can try the service without providing their own inputs?

File: service.py (hunk 5 above)
Comment (michaelfeil) [reply service.py:44]: Agreed, but this way generates an API client compatible to https://platform.openai.com/docs/guides/embeddings - therefore lets keep it like this.

File: bentofile.yaml (hunk 1 above)
Comment (michaelfeil) [reply bentofile.yaml]: removed the docker section.

**Review by larme — APPROVED** [review]
LGTM for now

**Pull request merged**